`test_result_type_numpy_func` assumes 64-bit architecture

The test `TestNumpyUnclassified.test_result_type_numpy_func` assumes it is running on a 64bit architecture.  This causes the test to fail on a 32bit architecture, I suspect because it expects an `int64` while numpy defaults to using an `int32`.  The full failure is below:

```
______________ TestNumpyUnclassified.test_result_type_numpy_func _______________

self = <pint.testsuite.test_numpy.TestNumpyUnclassified testMethod=test_result_type_numpy_func>

    @helpers.requires_array_function_protocol()
    def test_result_type_numpy_func(self):
>       self.assertEqual(np.result_type(self.q), np.dtype("int64"))
E       AssertionError: dtype('int32') != dtype('int64')
```

--- a/pint/testsuite/test_numpy.py
+++ b/pint/testsuite/test_numpy.py
@@ -885,7 +885,7 @@ class TestNumpyUnclassified(TestNumpyMethods):
 
     @helpers.requires_array_function_protocol()
     def test_result_type_numpy_func(self):
-        self.assertEqual(np.result_type(self.q), np.dtype("int64"))
+        self.assertEqual(np.result_type(self.q), np.dtype("int"))
 
     @helpers.requires_array_function_protocol()
     def test_nan_to_num_numpy_func(self):

